keras_cv/models/resnet_v2.py

```diff
@@ -94,214 +98,225 @@
+    x,
     filters,
     blocks,
     stride=2,
     dilations=1,
     name=None,
-    block_fn=Block,
+    block_type="block",
     first_shortcut=True,
     stack_index=1,
 ):
     """A set of stacked blocks.
+
     Args:
-        filters: integer, filters of the layer in a block.
-        blocks: integer, blocks in the stacked blocks.
-        stride: default 2, stride of the first layer in the first block.
-        name: string, stack label.
-        block_fn: callable, `Block` or `BasicBlock`, the block function to stack.
-        first_shortcut: default True, use convolution shortcut if True,
-          otherwise identity shortcut.
+        x: input tensor.
+        filters: int, filters of the layer in a block.
+        blocks: int, blocks in the stacked blocks.
+        stride: int, stride of the first layer in the first block. Defaults to 2.
+        dilation: int, the dilation rate to use for dilated convolution.
+            Defaults to 1.
+        block_type: string, one of "basic_block" or "block". The block type to
+            stack. Use "basic_block" for ResNet18 and ResNet34.
+        first_shortcut: bool. Use convolution shortcut if `True` (default),
+            otherwise uses identity or pooling shortcut, based on stride.
+
     Returns:
         Output tensor for the stacked blocks.
     """
+
     if name is None:
         name = f"v2_stack_{stack_index}"
 
-    def apply(x):
-        x = block_fn(filters, conv_shortcut=first_shortcut, name=name + "_block1")(x)
-        for i in range(2, blocks):
-            x = block_fn(filters, dilation=dilations, name=name + "_block" + str(i))(x)
-        x = block_fn(
-            filters,
-            stride=stride,
-            dilation=dilations,
-            name=name + "_block" + str(blocks),
-        )(x)
-        return x
+    if block_type == "basic_block":
+        block_fn = apply_basic_block
+    elif block_type == "block":
+        block_fn = apply_block
+    else:
+        raise ValueError(
+            """`block_type` must be either "basic_block" or "block". """
+            f"Received block_type={block_type}."
+        )
 
-    return apply
+    x = block_fn(x, filters, conv_shortcut=first_shortcut, name=name + "_block1")
+    for i in range(2, blocks):
+        x = block_fn(x, filters, dilation=dilations, name=name + "_block" + str(i))
+    x = block_fn(
+        x,
+        filters,
+        stride=stride,
+        dilation=dilations,
+        name=name + "_block" + str(blocks),
+    )
+    return x
 
 
-def ResNetV2(
-    stackwise_filters,
-    stackwise_blocks,
-    stackwise_strides,
-    include_rescaling,
-    include_top,
-    stackwise_dilations=None,
-    name="ResNetV2",
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classes=None,
-    classifier_activation="softmax",
-    block_fn=Block,
-    **kwargs,
-):
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ResNetV2(keras.Model):
```

Comment:
https://github.com/keras-team/keras-cv/issues/1015 !

Reply:
Thanks, I didn't see that Issue @innat! I hope you like this change.

I was piloting functional subclasses in this PR and will make a broader announcement soon.